Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/server/BaseService.java

```diff
@@ -64,8 +66,13 @@ protected AssertionType getAssertion(WebServiceContext context, AssertionType as
         }
 
         // Extract the message id value from the WS-Addressing Header and place it in the Assertion Class
-        if (assertion != null && NullChecker.isNullish(assertion.getMessageId())) {
-            assertion.setMessageId(extractor.getOrCreateAsyncMessageId(context));
+        if (assertion != null) {
```

Comment:
Could be possible that SAML2AssertionExtractor.getInstance().extractSamlAssertion(context) returns null (line 63).  SonarQube should have complained if its not possible to be null in line 69.